overlord/snapstate/storehelpers.go

```diff
@@ -82,6 +84,118 @@ func refreshOptions(st *state.State, origOpts *store.RefreshOptions) (*store.Ref
 	return &opts, nil
 }
 
+// installSizeInfo returns total download size of snaps and their prerequisites
+// (bases and default content providers), querying the store as neccessarry,
+// potentially more than once. It assumes the initial list of snaps already has
+// download infos set.
+// The state must be locked by the caller.
+func installSizeInfo(st *state.State, snaps []*snap.Info, userID int) (uint64, error) {
+	curSnaps, err := currentSnaps(st)
+	if err != nil {
+		return 0, err
+	}
+
+	user, err := userFromUserID(st, userID)
+	if err != nil {
+		return 0, err
+	}
+
+	accountedSnaps := map[string]bool{}
+	for _, snap := range curSnaps {
+		accountedSnaps[snap.InstanceName] = true
+	}
+
+	var prereqs []string
+
+	resolveBaseAndContentProviders := func(snapInfo *snap.Info) {
+		if snapInfo.SnapType != snap.TypeApp {
+			return
+		}
+		if snapInfo.Base != "none" {
+			base := defaultCoreSnapName
+			if snapInfo.Base != "" {
+				base = snapInfo.Base
+			}
+			if !accountedSnaps[base] {
+				prereqs = append(prereqs, base)
+				accountedSnaps[base] = true
+			}
+		}
+		for _, snapName := range defaultContentPlugProviders(st, snapInfo) {
+			if !accountedSnaps[snapName] {
+				prereqs = append(prereqs, snapName)
+				accountedSnaps[snapName] = true
+			}
+		}
+	}
+
+	snapSizes := map[string]uint64{}
+	for _, snapInfo := range snaps {
+		if snapInfo.DownloadInfo.Size == 0 {
+			// XXX: we could support this by simply adding to prereqs
```

Comment:
I would remove this comment, it's true only for the install case, but this function I think can be used as is also for refreshes, but for that case then this comment isn't true

Reply:
Maybe I'm confused, but is it ok to error out then (this is why I put that comment here)?

Reply:
yes, it is ok to error out, the doc comment explains that,  there's no way to do this correctly except by doing the exact same store request as the caller.

Also, now that I notice, the function should probably not have Info in the name

Reply:
Done.